### PR TITLE
feat: add query complexity validations

### PR DESCRIPTION
**Pull request by lgobbi-atix**
# Context

As noted in the [wiki](https://github.com/input-output-hk/cardano-graphql/wiki/Security#depth-limit), adding better query cost calculations would can be used to implement rate limiting or to reject complex queries and help prevent DoS attacks.

# Proposed Solution
Implement [graphql-query-complexity](https://github.com/slicknode/graphql-query-complexity) in the `server` and `api-cardano-db-hasura` package

# Important Changes Introduced
- `buildSchema` function in `api-cardano-db-hasura` now can receive an additional optional argument with custom query and field costs.
- `maxQueryComplexity` configuration was added in the `server` package to limit query complexity
- A `queryComplexityValidator` was added as a validation rule in `server`

### CODE REVIEW

File: packages/api-cardano-db-hasura/src/typeAliases.ts [hunk 1]
```diff
@@ -1,3 +1,19 @@
+import { ComplexityEstimator } from 'graphql-query-complexity'
```
Comment (rhyslbw) [inline packages/api-cardano-db-hasura/src/typeAliases.ts:1]: Best to put this into a separate file named `queryComplexity`. `typeAliases` is just meant for conveniences such as more specific types the omit or pick fields.

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Thanks @lgobbi-atix, this looks like a good extension to the existing code. Can you please add a few tests in `server/test` to demonstrate the behaviour?

**Review by rhyslbw — DISMISSED** [review]

**Review by rhyslbw — APPROVED** [review]
Really nice work here @lgobbi-atix.

**Comment by CyberCyclone** [comment]
Could we please have some documentation on how to set the environment variable for it to edit the value? The default 200 seems to small for my queries and setting maxQueryComplexity=1000 in the environment section of docker-compose.yml under graphql doesn't change it. Not sure if it's hooked up for that.

**Comment by rhyslbw** [comment]
Sorry @CyberCyclone, I missed your response. The released version reverted this feature to being opt-in, but if you are looking to configure the server, add `MAX_QUERY_COMPLEXITY` here:

https://github.com/input-output-hk/cardano-graphql/blob/754d9e9acc31a75b3f786cac5553bd3155e6a968/docker-compose.yml#L104-L107

**Pull request merged**